cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -189,11 +180,13 @@ public void run() {
           LOG.warn("Skipping goal violation detection for {} because load completeness requirement is not met.", goal);
         }
       }
-      if (!goalViolations.violatedGoalsByFixability().isEmpty()) {
+      Map<Boolean, List<String>> violatedGoalsByFixability = goalViolations.violatedGoalsByFixability();
+      if (!violatedGoalsByFixability.isEmpty()) {
         _anomalies.add(goalViolations);
       }
+      refreshBalancednessScore(violatedGoalsByFixability);
```

Comment:
If there is dead broker in the cluster, we will not reaches here, looks like currently this balancedness score(in `state` endpoint and the metric) works only under scenario there is not dead broker?

Reply:
Good catch -- it should be `0`!

#### Balancedness under Unhealthy Cluster State
A cluster is considered as unhealthy if it contains offline logDirs with replicas. Hence, depending on the replication factor of the corresponding partitions, minISR configuration of the topic, and the client's consistency requirements (e.g. acks=all), clients that produce to, or consume from the corresponding partitions with offline replicas are likely to experience undesired side-effects (e.g. higher latency, data loss). To indicate the severity of the issue, such clusters have "0" balancedness score. Note that similar approach does not apply to on-demand requests, because user-initiated requests  (1) allow manually marking healthy brokers as dead for forcing the removal of replicas from healthy brokers/disks, (2) evaluate the balancedness of each goal cumulatively (i.e. based on the incrementally optimized cluster state) rather than individually (i.e. based on the same initial cluster state) for optimization.